mlflow/dspy/load.py

```diff
@@ -37,17 +37,21 @@ def _set_tracer_context(model_path):
 def _load_model(model_uri, dst_path=None):
     import dspy
 
-    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
-    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name="dspy")
+    local_model_path = _download_artifact_from_uri(
+        artifact_uri=model_uri, output_path=dst_path
+    )
+    flavor_conf = _get_flavor_configuration(
+        model_path=local_model_path, flavor_name="dspy"
+    )
 
     _add_code_from_conf_to_system_path(local_model_path, flavor_conf)
     model_path = flavor_conf.get("model_path", _DEFAULT_MODEL_PATH)
     with open(os.path.join(local_model_path, model_path), "rb") as f:
         loaded_wrapper = cloudpickle.load(f)
+
+    _set_tracer_context(local_model_path, loaded_wrapper.dspy_settings["callbacks"])
     # Set the global dspy settings and return the dspy wrapper.
     dspy.settings.configure(**loaded_wrapper.dspy_settings)
```

Comment:
Q: Is there any reason for changing this ordering?

Reply:
yes, but pretty minor - I want all modifications to the loaded dspy settings to happen before `dspy.settings.configure` call for readability.